pages/p/[productId].js

```diff
@@ -120,6 +120,22 @@ const Product = React.memo(lazyProps => {
     </Row>
   )
 
+  function fetchVariant() {
+    fetch(`/api/p/${product.id}?color=${color.id}&size=${size.id}`)
+      .then(res => res.json())
+      .then(data => {
+        updateStore({
+          ...store,
+          pageData: { ...store.pageData, ...data },
```

Comment:
When you want to unselect color, it doesnt work. The tricky part is here that we check the equality of colors, to uncheck it, but because we created new object, the previous color is not equal to the new one, altho actual data it is. So we need to figure out what is the best way to solve this problem. One option would be to use lodash isEqual inside ProductOption, which solves the problem, but I am not sure if it is the best one.

Reply:
You can check again, but I believe I have fixed this.